docs/features/network-security-controls/network-policy.md

```diff
@@ -1,25 +1,141 @@
 
-By default, the network traffic from and to K8s pods is not restricted in any way. Using NetworkPolicy is a way to enforce network isolation of selected pods. When a pod is selected by a NetworkPolicy allowed traffic is specified by the `Ingress` and `Egress` sections.  
+Network Policy is a resource, which helps enforce the network isolation of pods. By default, the network traffic from and to K8s pods are not restricted in any way: NetworkPolicy resources allow us to enable and/or disable specific dataflows at a pod level.
+
+## Motivation
 
-Each NetworkPolicy object consists of four sections:
+The goal behind this feature is to create application or namespace scope policies that regulate intertenant communication. 
 
-1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
-2. `policyTypes`: determines which policy types are included, if none are selected then `Ingress` will always be set and `Egress` will be set if any `Egress` rules are applied 
-3. `Ingress rules`: determines the sources that pods selected by the ingress rule can receive traffic from 
-4. `Egress rules`: determines the sinks that pods selected by the egress rule can send trafic to 
+This feature is intended to allow application developers to secure their applications and namespaces using namespace scoped Network Policies.
 
-# NetworkPolicy features 
+This feature is not intended for Network Administrators, who should refer to AdminNetworkPolicies for higher level cluster scoped policies.
 
-These are described in order and are additive 
+## How to enable this feature on an OVN-Kubernetes cluster?
 
-## **Unicast default-deny**
+Feature is always enabled and triggered upon creation of NetworkPolicy objects.
 
-When a pod is selected by one or more NetworkPolicies, the `policyTypes` is set to both `Ingress` and `Egress`, and if no rules are specified it becomes isolated and all unicast ingress and egress traffic is blocked for pods in the same namespce as the NetworkPolicy. 
+## Workflow Description
+
+### Anatomy of a NetworkPolicy
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: sample-network-policy
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - from:
+    - namespaceSelector:
+      matchLabels:
+        project: myproject
+  egress:
+  - to:
+    - namespaceSelector:
+      matchLabels:
+        project: myproject
+
+```
+Each NetworkPolicy object (example above) consists of four sections:
+
+1. `podSelector`: a mandatory label selector that determines which pods the NetworkPolicy applies to
+2. `policyTypes`: indicates whether NetworkPolicy applies to `Ingress` or `Egress` traffic. `Egress` will be set if any `Egress` rules are applied, and vice versa if any `Ingress` rules are applied. If none are specified then `Ingress` will always be set.
```

Comment:
I would say it's not "vise versa", but more "respectively".

Reply:
"If none are specified then `Ingress` will always be set."
can you please clarify none of what?